drivers/serial/uart_sam.c

```diff
@@ -49,7 +49,7 @@ static int uart_sam_poll_in(const struct device *dev, unsigned char *c)
 	Uart * const uart = cfg->regs;
 
 	if (!(uart->UART_SR & UART_SR_RXRDY)) {
-		return -EBUSY;
+		return -1;
```

Comment:
Hi @javlands ,

You mention on the commit message that flag is wrong interpreted. So, what is the correct interpretation once -1 don't help much to understand the correct meaning? Can you clarify what is wrong, exactly?

Reply:
From `include/zephyr/drivers/uart.h`:

```c
/**
 * @brief Read a character from the device for input.
 *
 * This routine checks if the receiver has valid data.  When the
 * receiver has valid data, it reads a character from the device,
 * stores to the location pointed to by p_char, and returns 0 to the
 * calling thread. It returns -1, otherwise. This function is a
 * non-blocking call.
 *
 * @param dev UART device instance.
 * @param p_char Pointer to character.
 *
 * @retval 0 If a character arrived.
 * @retval -1 If no character was available to read (i.e. the UART
 *            input buffer was empty).
 * @retval -ENOSYS If the operation is not implemented.
 * @retval -EBUSY If async reception was enabled using @ref uart_rx_enable
 */
__syscall int uart_poll_in(const struct device *dev, unsigned char *p_char);
```

Which states that `-1` has to be returned in case no data was available,